server/server.go

```diff
@@ -0,0 +1,65 @@
+package server
+
+import (
+	"net/http"
+	"github.com/gruntwork-io/health-checker/options"
+	"net"
+	"fmt"
+)
+
+func StartHttpServer(opts *options.Options) {
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		checkTcpPorts(w, r, opts)
+	})
+	http.ListenAndServe(opts.Listener, nil)
+}
+
+// Check that we can open a TPC connection to all the ports in opts.Ports
+func checkTcpPorts(w http.ResponseWriter, r *http.Request, opts *options.Options) {
+	logger := opts.Logger
+	logger.Infof("Received inbound request. Beginning health checks...")
+
+	allPortsValid := true
+
+	for _, port := range opts.Ports {
+		err := attemptTcpConnection(port, opts)
+		if err != nil {
+			logger.Warnf("TCP connection to port %d FAILED!", port)
+			allPortsValid = false
+		} else {
+			logger.Infof("TCP connection to port %d successful", port)
+		}
+	}
+
+	if allPortsValid {
+		logger.Infof("All health checks passed. Returning HTTP 200 response.")
+		writeHttp200Response(w)
+	} else {
+		logger.Infof("At least one health check failed. Returning HTTP 504 response.")
+		writeHttp504Response(w)
```

Comment:
Writing to a response object leads to messy bugs. For example, you might call `writeXXXResponse` twice by accident (e.g., you forgot a `return`), and that'll lead to confusing, hard to debug issues.

A better approach is to return a `Response` type of some sort from this method that contains fields for `Status` and `Body`. The main method can then write this response in a single place. This also makes writing tests much easier.

Reply:
I like the "defensive programming" philosophy. Good call. Updated in 9645115c55430f0ad20fa55c09bdcad3a8b12f8f.